tests/phpunit/CRM/Member/Form/MembershipTest.php

```diff
@@ -1076,11 +1112,39 @@ public function testSubmitUpdateMembershipFromPartiallyPaid(): void {
     $form = $this->getContributionForm($submitParams);
     $form->postProcess();
 
-    //Check if Membership is updated to New.
+    // Check if Membership is updated to New.
     $membership = $this->callAPISuccessGetSingle('Membership', ['contact_id' => $this->_individualId]);
     $this->assertEquals($membership['status_id'], array_search('New', $memStatus));
   }
 
+  /**
+   * Test if membership is updated to New after contribution
+   * is updated from Partially paid to Completed.
+   *
+   * @throws \CRM_Core_Exception
+   */
+  public function testSubmitSaveMembershipNoChangesUnpaid(): void {
+
+    // Perform a pay later membership contribution.
+    $this->testSubmitPayLaterWithBilling();
+    $membership = $this->callAPISuccessGetSingle('Membership', ['contact_id' => $this->_individualId]);
+    $this->assertEquals(CRM_Core_PseudoConstant::getKey('CRM_Member_BAO_Membership', 'status_id', 'Pending'), $membership['status_id']);
+
+    $_REQUEST['id'] = $membership['id'];
```

Comment:
I'm not sure if this gets reset anywhere but it doesn't seem to break anything. I might put up a followup just to clear it at the end of the test to avoid future weirdness.